Fw/Dp/DpContainer.cpp

```diff
@@ -127,42 +127,110 @@ void DpContainer::serializeHeader() {
 void DpContainer::setBuffer(const Buffer& buffer) {
     // Set the buffer
     this->m_buffer = buffer;
-    // Check that the buffer is large enough to hold a data product packet
-    FW_ASSERT(buffer.getSize() >= MIN_PACKET_SIZE, buffer.getSize(), MIN_PACKET_SIZE);
+    // Check that the buffer is large enough to hold a data product packet with
+    // zero-size data
+    const FwSizeType bufferSize = buffer.getSize();
+    FW_ASSERT(bufferSize >= MIN_PACKET_SIZE, static_cast<FwAssertArgType>(bufferSize),
+              static_cast<FwAssertArgType>(MIN_PACKET_SIZE));
     // Initialize the data buffer
     U8* const buffAddr = buffer.getData();
     const FwSizeType dataCapacity = buffer.getSize() - MIN_PACKET_SIZE;
     // Check that data buffer is in bounds for packet buffer
-    FW_ASSERT(DATA_OFFSET + dataCapacity <= buffer.getSize());
+    const FwSizeType minBufferSize = DATA_OFFSET + dataCapacity;
+    FW_ASSERT(bufferSize >= minBufferSize, static_cast<FwAssertArgType>(bufferSize),
+              static_cast<FwAssertArgType>(minBufferSize));
     U8* const dataAddr = &buffAddr[DATA_OFFSET];
     this->m_dataBuffer.setExtBuffer(dataAddr, dataCapacity);
 }
 
-void DpContainer::updateHeaderHash() {
-    Utils::HashBuffer hashBuffer;
+Utils::HashBuffer DpContainer::getHeaderHash() const {
+    const FwSizeType bufferSize = this->m_buffer.getSize();
+    const FwSizeType minBufferSize = HEADER_HASH_OFFSET + HASH_DIGEST_LENGTH;
+    FW_ASSERT(bufferSize >= minBufferSize, static_cast<FwAssertArgType>(bufferSize),
+              static_cast<FwAssertArgType>(minBufferSize));
+    const U8* const buffAddr = this->m_buffer.getData();
+    return Utils::HashBuffer(&buffAddr[HEADER_HASH_OFFSET], HASH_DIGEST_LENGTH);
+}
+
+Utils::HashBuffer DpContainer::computeHeaderHash() const {
+    const FwSizeType bufferSize = this->m_buffer.getSize();
+    FW_ASSERT(bufferSize >= Header::SIZE, static_cast<FwAssertArgType>(bufferSize),
+              static_cast<FwAssertArgType>(Header::SIZE));
+    U8* const buffAddr = this->m_buffer.getData();
+    Utils::HashBuffer computedHash;
+    Utils::Hash::hash(buffAddr, Header::SIZE, computedHash);
+    return computedHash;
+}
+
+void DpContainer::setHeaderHash(Utils::HashBuffer hash) {
```

Comment:
Why not pass by reference?

Reply:
We could pass this one by const reference, but it's small, and I think it's a good idea to make the copy, since reading a serializable object updates the object.

Reply:
But if I make the change in the next comment, then we don't need the copy, because there's no destructive read.

Reply:
Fixed